tests/api/test_api.py

```diff
@@ -95,6 +95,46 @@ def test_is_node_schema_valid(cript_api: cript.API) -> None:
     assert cript_api._is_node_schema_valid(node_json=json.dumps(valid_file_dict)) is True
 
 
+def test_get_vocabulary_by_category(cript_api: cript.API) -> None:
+    """
+    tests if a vocabulary can be retrieved by category
+    1. tests response is a list of dicts as expected
+    1. tests that the fundamental identifiers exist within the API vocabulary response
+
+    tests that if a vocabulary is not valid category it raises an error
+
+    Warnings
+    --------
+    This test only gets the vocabulary category for "material_identifier_key" and does not test all the possible
+    CRIPT controlled vocabulary
+    """
+
+    material_identifier_vocab_list = cript_api.get_vocab_by_category("material_identifier_key")
+
+    # test response is a list of dicts
+    assert isinstance(material_identifier_vocab_list, list)
+
+    # test it has basic material identifiers
+    has_alternative_names = False
+    has_bigsmiles = False
+    has_smiles = False
+
+    for identifier in material_identifier_vocab_list:
```

Comment:
There should be a better way to test this.

something like `assert "bigsmiles" in material_identifier_list`.

Reply:
that is way way cleaner!

I'll add it in

Reply:
actually, I don't think I can do that because I it is a list of dicts

```json
[
   {
      "description":"preferred name",
      "name":"preferred_name",
      "validator":"None",
      "value_type":"string"
   },
   {
      "description":"additional names",
      "name":"alternative_names",
      "validator":"None",
      "value_type":"list[string]"
   },
   {
      "description":"Chemical ID",
      "name":"chemical_id",
      "validator":"None",
      "value_type":"string"
   },
   {
      "description":"bigSMILES line notation",
      "name":"bigsmiles",
      "validator":"None",
      "value_type":"string"
   },
   {
      "description":"simplified molecular-input line-entry system (SMILES)",
      "name":"smiles",
      "validator":"None",
      "value_type":"string"
   },
   {
      "description":"chemical formula",
      "name":"chem_formula",
      "validator":"None",
      "value_type":"string"
   },
   {
      "description":"chemical formula of repeating unit",
      "name":"chem_repeat",
      "validator":"None",
      "value_type":"list[string]"
   },
   {
      "description":"PubChem CID",
      "name":"pubchem_cid",
      "validator":"None",
      "value_type":"integer"
   },
   {
      "description":"IUPAC International Chemical Identifier",
      "name":"inchi",
      "validator":"None",
      "value_type":"string"
   },
   {
      "description":"a hashed version of the full InChI",
      "name":"inchi_key",
      "validator":"None",
      "value_type":"string"
   },
   {
      "description":"amino acid sequence",
      "name":"amino_acid",
      "validator":"None",
      "value_type":"string"
   },
   {
      "description":"company the material was obtain from",
      "name":"vendor",
      "validator":"None",
      "value_type":"string"
   },
   {
      "description":"lot or batch number",
      "name":"lot_number",
      "validator":"None",
      "value_type":"string"
   }
]
```

Reply:
do a list comprehension exactly you do it before in the actual code.
Then you can do simple `in` testing.

Reply:
agreed, that is cleaner

done! ✅